tutorials/05_plugin_loading.md

```diff
@@ -0,0 +1,158 @@
+mkdir -p ~/simple_loader/build
+cd simple_loader
+```
+
+Then download the example loader into your current directory by:
+
+```bash
+wget https://raw.githubusercontent.com/ignitionrobotics/ign-physics/main/examples/hello_world_loader/hello_world_loader.cc
+```
+
+### Examine the code
+
+At the top of the file `hello_world_loader.cc`, we include the headers that will
+be used in our code. After the `std` C++ libraries are the `Loader.hh` and
+`PluginPtr.hh`, which provides main functionalities for loading physics plugins
+and plugin pointers. Next includes from \ref ignition::physics are the tools for
+retrieving "Features" and "Entities" from physics plugins.
+
+```cpp
+#include <iostream>
+
+#include <ignition/plugin/Loader.hh>
+#include <ignition/plugin/PluginPtr.hh>
+
+#include <ignition/physics/FindFeatures.hh>
+#include <ignition/physics/GetEntities.hh>
+#include <ignition/physics/RequestEngine.hh>
+
+using Features = ignition::physics::FeatureList<
+    ignition::physics::GetEngineInfo
+>;
+```
+
+Next, in the main function, the loader requires users to provide a path for
+desired plugins to be loaded. The plugin names are retrieved by
+@ref ignition::plugin::Loader::LoadLib member function.
+
+```cpp
+if (argc <= 1)
+{
+  std::cerr << "Please provide the path to an engine plugin." << std::endl;
+  return 1;
+}
+
+std::string pluginPath = argv[1];
+
+ignition::plugin::Loader pl;
+auto plugins = pl.LoadLib(pluginPath);
+```
+
+Assuming the correct path, our loader will instantiate all plugins that are
+available in the path using @ref ignition::plugin::Loader::Instantiate member function. Then for each instantiated plugin, it will request an engine
+using @ref ignition::physics::RequestEngine3d<Features>::From function.
+
+```cpp
+for (const std::string &name : pluginNames)
+{
+  std::cout << "Testing plugin: " << name << std::endl;
+  ignition::plugin::PluginPtr plugin = pl.Instantiate(name);
+
+  auto engine = ignition::physics::RequestEngine3d<Features>::From(plugin);
+
+  std::cout << "  engine name: " << engine->GetName() << std::endl;
+}
+```
+
+### Example setup CMakeLists.txt file for Dome release
+
+Now create a file named `CMakeLists.txt` with your favorite editor and add these
+lines for finding `ign-plugin` and `ign-physics` dependencies in Dome release:
+
+```cmake
+cmake_minimum_required(VERSION 3.5 FATAL_ERROR)
+
+set(IGN_PLUGIN_VER 1)
+find_package(ignition-plugin${IGN_PLUGIN_VER} 1.1 REQUIRED COMPONENTS all)
+
+set(IGN_PHYSICS_VER 3)
+find_package(ignition-physics${IGN_PHYSICS_VER} REQUIRED)
+```
+
+After that, add the executable pointing to our file and add linking library so
+that `cmake` can compile it:
+
+```cmake
+add_executable(hello_world_loader hello_world_loader.cc)
+target_link_libraries(hello_world_loader
+  ignition-plugin${IGN_PLUGIN_VER}::loader
+  ignition-physics${IGN_PHYSICS_VER}::ignition-physics${IGN_PHYSICS_VER})
+```
+
+## Build and run
+
+### Compile the loader
+
+Now you can build the loader by:
```

Comment:
Would be nice to point out the layout of folder structure and where to run `cmake` commands